lib/PrintAsClang/DeclAndTypePrinter.cpp

```diff
@@ -949,16 +968,35 @@ class DeclAndTypePrinter::Implementation
     DeclAndTypeClangFunctionPrinter funcPrinter(os, owningPrinter.prologueOS,
                                                 owningPrinter.typeMapping,
                                                 owningPrinter.interopContext);
+    llvm::SmallVector<DeclAndTypeClangFunctionPrinter::AdditionalParam, 2>
+        additionalParams;
+    auto ABIparams = owningPrinter.interopContext.getIrABIDetails()
+                         .getFunctionABIAdditionalParams(FD);
+    if (funcTy->isThrowing() && !ABIparams.empty()) {
+      if (ABIparams.pop_back_val().role ==
+          IRABIDetailsProvider::ABIAdditionalParam::ABIParameterRole::Self)
+        additionalParams.push_back(
```

Comment:
Can you share this code for constructing additional params with the section above where you create them too for C function?

Reply:
I refactored it by pushing this piece of code to a separated function and replaced the code with a call to this function